drivers/SmartThings/philips-hue/src/hue/api.lua

```diff
@@ -267,28 +270,39 @@ function PhilipsHueApi.request_api_key(bridge_ip, socket_builder)
   return table.unpack(recv, 1, recv.n)
 end
 
-function PhilipsHueApi:get_lights() return do_get(self, "/clip/v2/resource/light") end
+function PhilipsHueApi:get_all_reprs_for_rtype(rtype) return do_get(self, string.format("/clip/v2/resource/%s", string.lower(tostring(rtype)))) end
+
+function PhilipsHueApi:get_rtype_by_rid(rtype, rid)
+  return do_get(
+    self,
+    string.format("/clip/v2/resource/%s/%s",
+    string.lower(tostring(rtype)),
+    string.lower(tostring(rid)))
```

Comment:
I might add an indent here to indicate these are arguments to `string.format` and not `do_get`. I was a bit confused for a sec.